addons/mass_mailing/models/mailing.py

```diff
@@ -1326,7 +1328,13 @@ def _image_to_url(self, b64image: bytes):
             attachment.id, attachment.access_token)
 
     def _get_image_from_url(self, url):
-        return requests.get(url).content
+        head = requests.head(url, allow_redirects=True, timeout=5)
+        maxsize = int(tools.config.get("import_image_maxbytes", DEFAULT_IMAGE_MAXBYTES))
+        if (
+            int(head.headers.get('Content-Length', 0)) <= maxsize and
+            'image' in head.headers.get('Content-Type', 'text/html')
```

Comment:
@mart-e Thanks for the feedback, makes sense. Does this work for you? If so I'll squash it with the commit where it's introduced.